### ISSUE TITLE
(Set a clear title describing your idea)

### ISSUE THREAD

**Issue by slifty** [slackapi/node-slack-sdk#1768]
I'm using `@slack/socket-mode` in a TypeScript project and would love to be able to benefit from type definitions when defining event handlers -- e.g. when I invoke something along the lines of

```
socketModeClient.on(
  'reaction_added',
  async (event) => {
    ...
  }
);
```

I'd like `event` to not be of the `any` type.

It looks like there are [libraries](https://github.com/andywer/typed-emitter) that might support this, for your consideration.

#### Packages:

Select all that apply:

- [ ] `@slack/web-api`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [x] `@slack/socket-mode`
- [ ] `@slack/types`
- [ ] I don't know

### Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by filmaj** [comment]
I agree this would be a fantastic enhancement. The issue is that perfect typing for Slack payload events is challenging, as the payloads can be quite contextual. For example, depending on the Slack workspace type (enterprise vs. standalone), the payloads might differ. Additionally, some payloads may or may not contain properties depending on the context, for example 
a [`block_actions` payload contains a `container` property](https://api.slack.com/reference/interaction-payloads/block-actions?ref=bk#fields), which might be a `View` object or might be a containing message.

That said, at least some baseline form of event typing should be doable. I will leave this as an enhancement issue open but can make no promises on a delivery timeline.
